keyboards/kmac/kmac.h

```diff
@@ -13,41 +13,57 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-#ifndef KMAC_H
-#define KMAC_H
+#pragma once
 
 #include "quantum.h"
 
-// Keymap for the winkey version of the PCB.
-#define LAYOUT(                                                                                                                           \
-    K00, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E, K0F, K0G,                                                       \
-    K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E, K1F, K1G,                                                  \
-    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, K2E, K2F, K2G,                                                  \
-    K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3D,                                                                      \
-    K40, K41, K42, K43, K44, K45, K46, K47, K48, K49, K4A, K4D, K4F,                                                                      \
-    K50, K51, K52, K55, K58, K5A, K5C, K5D, K5E, K5F, K5G)                                                                                \
-    {                                                                                                                                     \
-        /*        0      1      2      3      4      5      6      7      8      9      A      B      C      D      E      F      G    */ \
-        /* 0 */ {K00, KC_NO, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E, K0F, K0G},                                  \
-            /* 1 */ {K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E, K1F, K1G},                                \
-            /* 2 */ {K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, K2E, K2F, K2G},                                \
-            /* 3 */ {K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, KC_NO, K3D, KC_NO, KC_NO, KC_NO},                        \
-            /* 4 */ {K40, K41, K42, K43, K44, K45, K46, K47, K48, K49, K4A, KC_NO, KC_NO, K4D, KC_NO, K4F, KC_NO},                        \
-        /* 5 */ { K50, K51, K52, KC_NO, KC_NO, K55, KC_NO, KC_NO, K58, KC_NO, K5A, KC_NO, K5C, K5D, K5E, K5F, K5G }                       \
-    }
+#define LAYOUT_all( \
```

Comment:
Because `LAYOUT_all` and `LAYOUT_tkl_ansi` are the same, you could actually replace Lines 20-35 with:

```c
#define LAYOUT_all LAYOUT_tkl_ansi
```

... and delete the `LAYOUT_all` object from `info.json` and the whole thing would still work as intended (though perhaps moving the short-form definition later in the file would be less confusing/more accessible to novices).

This is completely optional though; don't feel obligated to do this.

Reply:
I did things this way due to this section in the documentation https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=ltkeyboard_namehgt. Which states that I should have a `LAYOUT_all` and use that for the default keymap.
I was thinking that it was required by the configurator or something like that. But maybe it is only necessary in the `info.json`?

I would prefer to not have the duplication as it is a bit confusing, exactly as you state. But I'm not sure if it is possible. If the `LAYOUT_all` is indeed required and I need the `LAYOUT_tkl_ansi` for the layout feature (or at least I believe I do).

Reply:
Perhaps `#define LAYOUT_tkl_ansi LAYOUT_all` then? The Community Layouts feature only needs the macro to compile; it doesn't matter how.

Reply:
That's a good idea 👍. I guess I still need the duplicates in `keymaps` and layouts in `info.json` but it's definitely an improvement.

Reply:
> I guess I still need the duplicates in `keymaps` and layouts in `info.json` but it's definitely an improvement.

I've forgotten on probably four separate occasions which keys in `info.json` are needed for this implementation. I plan on figuring this out in the near term and actually writing it down this time. :laughing: 

Having duplicate keymaps isn't really an issue; it's the most obvious place for a user to look to learn how to use their desired layout macro.

Reply:
> I've forgotten on probably four separate occasions which keys in info.json are needed for this implementation. I plan on figuring this out in the near term and actually writing it down this time.

I did my best to follow what is documented regarding the [info.json]( https://docs.qmk.fm/#/reference_info_json?id=infojson). But I have to admit that there were quite a bit of guessing involved on my end. So any improvements to the documentation regarding these things would definitely be appreciated.

Reply:
If LAYOUT_all and LAYOUT_tkl_ansi are the same, I would prefer we just stick with LAYOUT_tkl_ansi and abandon LAYOUT_all. LAYOUT_all should really only be used on a LAYOUT macro that supports "all" possible layouts.

Reply:
That's fine by me, though, as stated above, it appears to run a bit contrary to the documentation.

I have removed both the `LAYOUT_all` macro and the entry in `info.json`.